andrepereiradasilva/joomla-cms
# modals for menus (3.7x branch)

Pull Request for Improvement.
### Summary of Changes

This PR is the sequence of https://github.com/joomla/joomla-cms/pull/11830 but for now for menus modal in **3.7.x branch** .

For list of changes see https://github.com/joomla/joomla-cms/pull/11830 (same changes, but for menus in 3.7.x branch now).

Similiar PR: 
- Article: https://github.com/joomla/joomla-cms/pull/11830
- Category: https://github.com/joomla/joomla-cms/pull/11857
- Contact: https://github.com/joomla/joomla-cms/pull/11858
- Newsfeed: https://github.com/joomla/joomla-cms/pull/11862
- Menus (3.7.x): this one
### Animated Gif (click to view in full screen)

![menus-modal](https://cloud.githubusercontent.com/assets/9630530/18228447/629fd5dc-7246-11e6-93c3-ebfb9e6384bd.gif)
### Testing Instructions
- Use joomla latest 3.7.x branch
- Apply this patch
- Create a Menu item Alias Type
- Test all the "Edit", "Create", "Select" and "Clear" buttons  (as animated gif)
- In a multilanguage install edit a menu and try the modal button in the associations tab
- Test all the "Edit", "Create", "Select" and "Clear" buttons.
- Code review to check all is ok.

Note: there are more sites when this modal change is apllied (see code diff), you could test them too: mod_login, mod_menu, logout menu item, login menu item
### Documentation Changes Required

None.
